File: Perl/3-DeLaCruzAngel-Burbuja.pl
@vect;
$cantnum;
$aux;
$flg;
$ctr;
#-cantnum:La cantidad de numeros que el usuario quiere introducir
#-vect:Es el vector donde se van a ordenar los numeros
#-aux:es una variable temporal que guarda uno de los valors que van a cambiar de lugar
#-ctr: contador que delimita los limites del for
#-flg: Bandera que indica el estado de orden y desorden del vector
print "Escriba la cantidad de numeros a ordenar, este programa acepta hasta cien numeros en rango de -20000 a 20000\n";
$cantnum=<STDIN>; #Se escanea la cantidad de numeros a ordenar
if($cantnum>100)
{
 print "El programa no acepta mas de cien numeros"	
}
print "Ingrese los numeros a ordenar, despues de cada numero presione ENTER\n";
for($ctr=0; $ctr<$cantnum; $ctr++) #El limite superior es la cantidad de numeros que se va a ordenar menos uno
#Se le resta uno porque el vector empieza en la posicion cero
{
 $vect[$ctr]=<STDIN>;#Se guardan los numeros en el vector
}
$flg=0; #La bandera inicia en cero, que indica que el vector esta en desorden
while($flg==0)#Mientras el vector este desordenado el ciclo se sigue repitiendo
{
 for($ctr=0; $ctr<$cantnum-1; $ctr++)#El limite superior es la cantidad de numeros menos uno porque al llegar al final
  #el numero que esta en la ultima posicion no puede ser comparado con nada, pues no hay nada adelante de el
 {
  $flg=1; #La bandera se vuelve positiva, que indica que esta ordenado, y si la condicion de abajo no se cumple sale del ciclo while
  if($vect[$ctr]>=$vect[$ctr+1])#Si la posicion del vector es mayor que la posicion siguiente se efectua el intercambio de lugares
  {
    $aux=$vect[$ctr];#La variable temporal aux toma el valor del numero mayor
   	$vect[$ctr]=$vect[$ctr+1]; #El numero menor pasa una posicion atras
   	$vect[$ctr+1]=$aux; #La variable temporal deposita el valor una casilla adelante de donde estaba originalmente
   	$flg=0;	#La bandera se vuelve cero pues el vector esta en desorden
  }
 }	
}
print "Los numeros ordenados de forma ascendente son\n";
foreach $aux (@vect)
{
 print "$aux"; #imprime los numeros de forma ascendente
}

File: C/3-DeLaCruzAngel-Burbuja.c
#include<stdio.h>
int main() 
{
 int cantnum=0, vect[1000], aux=0, ctr=0, flg=0;
 /*-cantnum:La cantidad de numeros que el usuario quiere introducir
   -vect:Es el vector donde se van a ordenar los numeros
   -aux:es una variable temporal que guarda uno de los valors que van a cambiar de lugar
   -ctr: contador que delimita los limites del for
   -flg: Bandera que indica el estado de orden y desorden del vector */
 printf("Escriba la cantidad de números a ordenar, este programa ordena numeros enteros con un rango entre -20000 y 20000 y soporta hasta cien numeros\n");
 scanf("%d", &cantnum);//Se escanea la cantidad de numeros que el usuario quiere ordenar
 if(cantnum>100)//Se valida que el usuario no pueda introducir mas de cien numeros
 {
  printf("Este programa solo acepta hasta cien numeros"); 
  return 0;	
 }
 printf("Ingrese los numeros que desea ordenar, despues de cada numero presione la tecla Enter");
 for(ctr=0; ctr<cantnum; ctr++)
 {
  scanf("%d", &vect[ctr]); //Se guardan en el vector los numeros a ordenar
 }
 while(flg==0) //La bandera en cero indica desorden, es decir, mientras que el vector este en desorden, el ciclo se repite
 {
  for(ctr=0; ctr<cantnum-1; ctr++) /*El limite superior es la cantidad de numeros menos uno porque al llegar al final
  el numero que esta en la ultima posicion no puede ser comparado con nada*/
  { 
    flg=1; //La bandera se vuelve uno, y si la condición de abajo no se cumple, sale del ciclo     
   if(vect[ctr]>=vect[ctr+1]) /*Se verifica que los numeros no sean mayores al inmediato siguiente, de forma que si son mayores o iguales
    se hace el cambio de posicion*/
   {
   	aux=vect[ctr]; //la variable temporal toma el valor del numero mayor
   	vect[ctr]=vect[ctr+1]; //El numero mayor cambia de lugar una posicion atras
   	vect[ctr+1]=aux; //La variable temporal asigna su valor una posicion adelante de la que originalmente tenia
   	flg=0; //La bandera se vuelve cero porque el vector esta desordenado y el ciclo se sigue haciendo
   }
  }
  }	
   printf("Los numeros ordenados de forma ascendente son\n");
 for(ctr=0; ctr<cantnum; ctr++)
 {
  printf("%d ", vect[ctr]);	//Se imprimen los numeros ordenados de forma ascendente
 }
 return 0; 
}